common/lib/common-utils/.eslintrc.js

```diff
@@ -5,16 +5,55 @@
 
 module.exports = {
     "extends": [
-        "@fluidframework/eslint-config-fluid"
+        "@fluidframework/eslint-config-fluid/recommended"
     ],
     "parserOptions": {
-        "project": [ "./tsconfig.json", "./src/test/mocha/tsconfig.json", "./src/test/jest/tsconfig.json", "./src/test/types/tsconfig.json" ]
+        "project": [
+            "./tsconfig.json",
+            "./src/test/mocha/tsconfig.json",
+            "./src/test/jest/tsconfig.json",
+            "./src/test/types/tsconfig.json",
+        ]
     },
     "rules": {
         "@typescript-eslint/no-non-null-assertion": "off",
         "@typescript-eslint/strict-boolean-expressions": "off",
+        "@typescript-eslint/no-implied-eval": "error",
+        "@typescript-eslint/promise-function-async": "error",
+        "default-case": "error",
         "no-bitwise": "off",
         "no-null/no-null": "off",
-        "prefer-rest-params": "off"
-    }
+        "prefer-rest-params": "off",
+        "import/no-internal-modules": "error",
+        "import/no-nodejs-modules": [
```

Comment:
In my opinion, we should have a long-term goal to not rely on implicit polyfills. #8725 covers some of the reasons, and at the very least I think our lint config should help you understand the choice you're making. That said, I could see it being an exception to the "don't override rules at the project level" rule, because we do have some node-only packages in the server code.